enterprise/frontend/src/embedding-sdk/components/private/Breadcrumb/SdkBreadcrumbsProvider.tsx

```diff
@@ -0,0 +1,61 @@
+import { type ReactNode, createContext, useCallback, useState } from "react";
+
+import type {
+  SdkBreadcrumbItem,
+  SdkBreadcrumbsContextType,
+} from "embedding-sdk/types/breadcrumb";
+
+import { updateBreadcrumbsWithItem } from "./utils/update-breadcrumbs";
+
+export const SdkBreadcrumbsContext =
+  createContext<SdkBreadcrumbsContextType | null>(null);
+
+export interface SdkBreadcrumbsProviderProps {
+  children: ReactNode;
+}
+
+export const SdkBreadcrumbsProvider = ({
+  children,
+}: SdkBreadcrumbsProviderProps) => {
+  const [breadcrumbs, setBreadcrumbs] = useState<SdkBreadcrumbItem[]>([]);
+
+  const reportLocation = useCallback((item: SdkBreadcrumbItem) => {
+    setBreadcrumbs((prevBreadcrumbs) =>
+      updateBreadcrumbsWithItem(prevBreadcrumbs, item),
+    );
+  }, []);
+
+  const navigateTo = useCallback(
+    (breadcrumb: SdkBreadcrumbItem) => {
+      const existingIndex = breadcrumbs.findIndex(
+        (b) => b.id === breadcrumb.id && b.type === breadcrumb.type,
```

Comment:
Let's rename `b` to something non-single-character

Reply:
Good call, I missed that!

Reply:
Fixed in https://github.com/metabase/metabase/pull/62128/commits/0efdc6a607ad880913e9f01e0e63701f08db5e89